tfhe/src/core_crypto/algorithms/test/glwe_linear_algebra.rs

```diff
@@ -0,0 +1,776 @@
+use super::*;
+
+fn glwe_encrypt_add_assign_decrypt_custom_mod<Scalar: UnsignedTorus>(params: TestParams<Scalar>) {
+    let glwe_dimension = params.glwe_dimension;
+    let polynomial_size = params.polynomial_size;
+    let glwe_modular_std_dev = params.glwe_modular_std_dev;
+    let ciphertext_modulus = params.ciphertext_modulus;
+    let message_modulus_log = params.message_modulus_log;
+    let encoding_with_padding = get_encoding_with_padding(ciphertext_modulus);
+
+    let mut rsc = TestResources::new();
+
+    const NB_TESTS: usize = 10;
+    let msg_modulus = Scalar::ONE.shl(message_modulus_log.0);
+    let mut msg = msg_modulus;
+    let delta: Scalar = encoding_with_padding / msg_modulus;
+
+    while msg != Scalar::ZERO {
+        msg = msg.wrapping_sub(Scalar::ONE);
+        for _ in 0..NB_TESTS {
+            let glwe_sk = allocate_and_generate_new_binary_glwe_secret_key(
+                glwe_dimension,
+                polynomial_size,
+                &mut rsc.secret_random_generator,
+            );
```

Comment:
This could be in a common generic functions which take a closure as input

Reply:
not really, parameters required per tests vary and we don't always query all of them, so it kind of defeats the purpose of the generic function ?

Reply:
I'm not sure

Reply:
Not sure either, gotta test to know :man_shrugging:

Reply:
I'm a bit wary about genericizing test code, it was painful in concrete-core, and sometimes when the test code is not fully laid out it's a pain to debug, so unless you think it's absolutely necessary I'm not too kin on changing that and am fine with repeating myself in the tests